docs/content/docs/dev/table/tuning.md

```diff
@@ -368,3 +368,52 @@ FROM TenantKafka t
+
+This feature is currently enabled by default and regular join will be optimized into delta join when the following conditions are simultaneously met:
+
+1. The sql pattern satisfies the optimization criteria. For details, please refer to [Supported Features and Limitations]({{< ref "docs/dev/table/tuning" >}}#supported-features-and-limitations)
+2. The external storage system of the source table provides index information for fast querying for delta joins. Currently, [Apache Fluss(Incubating)](https://fluss.apache.org/blog/fluss-open-source/) has provided index information at the table level for Flink, allowing such tables to be used as source tables for delta joins. Please refer to the [Fluss documentation](https://fluss.apache.org/docs/0.8/engine-flink/delta-joins/#flink-version-support) for more details.
+
+### Working Principle
+
+In Flink, regular joins require completely storing incoming data from both sides in the state, matching that data when the opposite side's data arrives. In contrast, delta joins utilize the indexing capabilities provided by external storage systems to convert the behavior of querying state data into efficient queries against data in the external storage system using index keys. This approach avoids the need for duplicate storage of the same data in both the external storage system and the state.
+
+{{< img src="/fig/table-streaming/delta_join.png" width="70%" height="70%" >}}
+
+### Important Configurations
+
+Currently, the optimization for delta joins is enabled by default. You can disable this feature manually by setting the following configuration. Please see [Configuration]({{< ref "docs/dev/table/config" >}}#optimizer-options) page for more details.
+
+```sql
+SET 'table.optimizer.delta-join.strategy' = 'NONE';
+```
+
+Additionally, you can adjust the performance of delta joins by configuring the following configurations. Please see [Configuration]({{< ref "docs/dev/table/config" >}}#execution-options) page for more details.
+
+- `table.exec.delta-join.cache-enabled`
+- `table.exec.delta-join.left.cache-size`
+- `table.exec.delta-join.right.cache-size`
+
+### Supported Features and Limitations
+
+Delta joins are continuously evolving, and supports the following features currently.
+
+1. Support INSERT-ONLY tables as source tables for delta join.
+2. Support CDC tables without DELETE as source tables for delta join.
+3. Support project and filter between source and delta join.
+4. Support cache in delta join.
+
+However, delta joins has the following limitations. Any job containing one of these conditions cannot be optimized into a delta join.
```

Comment:
nit: "However, Delta Joins also have several **limitations**.
Jobs containing any of the following conditions **cannot** be optimized into a delta join:

1. The **index key** of the table must be included in the join’s **equivalence conditions**.
2. Only **INNER JOIN** is currently supported.
3. The **downstream operator** must be able to handle **duplicate changes**, such as a sink operating in **UPSERT mode** without `upsertMaterialize`.
4. When consuming a **CDC stream**, the **join key** must be part of the **primary key**.
5. When consuming a **CDC stream**, all **filters** must be applied on the **upsert key**.
6. **Non-deterministic functions** are not allowed in filters or projections."